daemon/docker_watcher.go

```diff
@@ -249,6 +255,21 @@ func (d *Daemon) handleCreateContainer(id string, retry bool) {
 			}
 		}
 
+		if ep != nil {
```

Comment:
Just move this to below the `ep ==nil` condition so you don't have to check for `ep != nil`